### ISSUE TITLE
Freezing some of the parameters in SAEM in nlmixr

### ISSUE THREAD

**Issue by krinaj** [nlmixrdevelopment/nlmixr#17]
Hello Rik, Wenping, and all,

I am trying out saem in nlmixr and wondering about how to freeze some of the parameters. I am hoping to read in some parameters directly from the dataset, but it does not work. It runs, but kind of defaults to some theta in estimates (I know for sure that it is not correct estimate at all). Here is what I am trying. I have V1, V2, CL, and Q defined in the dataset (Sims3) (same estimates for all IDs).

ODE <- 
"C1 = centr/V1;
C2 = peri/V2;
d/dt(depot) = -Ka*depot;
d/dt(centr) = F1*Ka*depot - CL*C1 - Q*C1 + Q*C2;
d/dt(peri) = Q*C1 - Q*C2;
"
mypar <- function(lKa, lKdeg, V1, V2, CL, Q)
{Ka = exp(lKa)
F1=F1
 V1=V1
 V2=V2
 CL=CL
 Q=Q
}

Sims3$COV <- 1
m2 = RxODE(ODE, modName="m2")
PRED = function() C1
saem_fit <- gen_saem_user_fn(model=m2, PKpars=mypar, pred=PRED)
model = list(saem_mod=saem_fit, res.mod=3, covars="COV")
inits = list(theta=c(Ka=0.1))
cfg = configsaem(model, data=Sims3, inits)
cfg$print = 50
fit = saem_fit(cfg)
summary(fit)
plot(fit)

Instead if I switch mypar with something like following, it works and gives me reliable estimates.
mypar <- function(lKa, V1, V2, CL, Q)
{Ka = exp(lKa)
 V1=2
 V2=3
 CL=0.3
 Q=0.2
F1=0.65
}

Any way to define parameters that I want to estimate in dataset rather than in function?

Thanks,
Krina

**Comment by wwang-at-github** [comment]
Dear Krina:

This is an excellent & non-trivial question.  Let me put some thoughts into
it & get back to you.

Thanks for trying out nlmixr.

Cheers!
Wenping

**Comment by mattfidler** [comment]
Hi Krina,

This is a new feature of nlmixr 1.0:

```R
one.cmt <- function() {
    ini({
        tka <- .5   # log Ka
        tcl <- -3.2 # log Cl
        tv <- fix(-1)    # log V
        eta.ka ~ 1
        eta.cl ~ 2
        add.err <- 0.1
    })
    model({
        ka <- exp(tka + eta.ka)
        cl <- exp(tcl + eta.cl)
        v <- exp(tv)
        linCmt() ~ add(add.err)
    })
}

f <- nlmixr(one.cmt,theo_sd,"saem")
```
You can see it is fixed by printing out the fit:
```
> f
-- nlmixr SAEM(Solved); OBJF not calculated fit ------------------------------------------------------------------------------------------------------------
      OBJF AIC BIC Log-likelihood Condition Number
SAEMg   NA  NA  NA             NA         2986.735

-- Time (sec; f$time): -------------------------------------------------------------------------------------------------------------------------------------
  saem setup optimize covariance table
 41.82  14.5        0          0  0.02

-- Population Parameters (f$parFixed): ---------------------------------------------------------------------------------------------------------------------
        Parameter   Est.       SE  %RSE    Back-transformed(95%CI) BSV(CV%) Shrink(SD)%
tka        log Ka 0.2189 0.003712 1.695       1.245 (1.236, 1.254)   79.21%   -0.2018% 
tcl        log Cl -3.117  0.08486 2.722 0.04428 (0.03749, 0.05229)   30.89%    -4.254% 
tv          log V     -1    FIXED FIXED                     0.3679                     
add.err           0.8098                                    0.8098                     


  Covariance Type (f$covMethod): fim
  No correlations in between subject variability (BSV) matrix
  Full BSV covariance (f$omega) or correlation (f$omegaR; diagonals=SDs)
  Distribution stats (mean/skewness/kurtosis/p-value) available in f$shrink

-- Fit Data (object f is a modified tibble): ---------------------------------------------------------------------------------------------------------------
# A tibble: 132 x 17
  ID     TIME    DV  PRED     RES IPRED   IRES  IWRES eta.ka eta.cl    ka     cl     v  rx1c Central   rx0   rx1
* <fct> <dbl> <dbl> <dbl>   <dbl> <dbl>  <dbl>  <dbl>  <dbl>  <dbl> <dbl>  <dbl> <dbl> <dbl>   <dbl> <dbl> <dbl>
1 1     0      0.74  0     0.74    0     0.74   0.914  0.340 -0.705  1.75 0.0219 0.368  0       0     4.02  0   
2 1     0.25   2.84  2.88 -0.0365  3.84 -1.000 -1.23   0.340 -0.705  1.75 0.0219 0.368  3.84    3.84  2.60  1.41
3 1     0.570  6.57  5.34  1.23    6.76 -0.191 -0.236  0.340 -0.705  1.75 0.0219 0.368  6.76    6.76  1.48  2.49
# ... with 129 more rows
```

**Comment by mattfidler** [comment]
For SAEM you can only fix parameters without an associated ETA
